src/ast.rs

```diff
@@ -730,3 +734,50 @@ impl From<ScopedVariable> for Expression {
         Expression::Variable(variable.into())
     }
 }
+
+/// Attribute shorthands
+#[derive(Debug, Eq, PartialEq)]
+pub struct AttributeShorthands(HashMap<Identifier, AttributeShorthand>);
+
+impl AttributeShorthands {
+    pub fn new() -> Self {
+        Self(HashMap::new())
+    }
+
+    pub fn get(&self, name: &Identifier) -> Option<&AttributeShorthand> {
+        self.0.get(name)
+    }
+
+    pub fn add(&mut self, shorthand: AttributeShorthand) {
+        self.0.insert(shorthand.name.clone(), shorthand);
+    }
+
+    pub fn iter(&self) -> std::collections::hash_map::Values<Identifier, AttributeShorthand> {
```

Comment:
Can you make the return type opaque?  I think it would be something like

``` rust
pub fn iter(&self) -> impl Iterator<Item=&AttributeShorthand> + '_
```
or maybe

``` rust
pub fn iter(&self) -> impl Iterator<Item=&'_ AttributeShorthand> + '_
```

Reply:
Brilliant! I didn't know how to do this.